File: leetcode/1268/SearchSuggestionsSystem.java
import java.util.*;

class SearchSuggestionsSystem {
    public static void main(String[] args) {
        Solution sl = new Solution();
        List<List<String>> al = sl
                .suggestedProducts(new String[] { "mobile", "mouse", "moneypot", "monitor", "mousepad" }, "mouse");

        for (List<String> list : al) {
            System.out.print("[");
            for (String str : list) {
                // doSomething
                System.out.print(str + ",");
            }
            System.out.println("]");
        }
    }
}
 
class Solution {
    public List<List<String>> suggestedProducts(String[] products, String searchWord) {
        List<List<String>> retList = new ArrayList<List<String>>();
        Arrays.sort(products);
        for (int i = 0; i < searchWord.length(); i++) {
            List<String> listString = new ArrayList<String>();
            for (int j = 0; j < products.length; j++) {
                if (listString.size() < 3 && i < products[j].length()
                        && (products[j].substring(0, i + 1).equals(searchWord.substring(0, i + 1)))) {
                    listString.add(products[j]);
                }
            }
            // if (listString.size() > 0) {
                retList.add(listString);
                // }
        }
        return retList;
    }
}

File: leetcode/1268/SearchSuggestionsSystem.cs
class SearchSuggestionsSystem
{
    public static void Main(string[] args)
    {
        Solution sl = new Solution();
        sl.SuggestedProducts();
    }
}
public class Solution
{
    public IList<IList<string>> SuggestedProducts(string[] products, string searchWord)
    {
        IList<IList<string>> retList = new List<IList<string>>();
        Array.Sort(products);
        for (int i = 0; i < searchWord.Length; i++)
        {
            List<string> listString = new List<string>();
            for (int j = 0; j < products.Length; j++)
            {
                if (listString.Count < 3 && i < products[j].Length
                        && (products[j].Substring(0, i + 1).Equals(searchWord.Substring(0, i + 1))))
                {
                    listString.Add(products[j]);
                }
            }
            // if (listString.Count > 0) {
            retList.Add(listString);
            // }
        }
        return retList;
    }
}

